### ISSUE TITLE
Can't disable warnings for anonymous functions

### ISSUE THREAD

**Issue by HaleTom** [koalaman/shellcheck#1931]
#### For bugs
- Rule Id (if any, e.g. SC1000): 1072, 1073
- My shellcheck version (`shellcheck --version` or "online"): online 
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
#      ^----- get shellcheck hints based on bash
# https://github.com/koalaman/shellcheck/issues/809

# shellcheck disable=SC1073,1072
() {  # Anonymous function for local variables
  var=$XDG_DATA_HOME/zinit/bin/zinit.zsh
  do_something_with "$var"
}
```

#### Here's what shellcheck currently says:
```
$ shellcheck myscript
 
Line 6:
() {  # Anonymous function for local variables
^-- SC1073: Couldn't parse this explicit subshell. Fix to allow more checks.
   ^-- SC1072: Unexpected keyword/token. Fix any mentioned problems and try again.

$
```

#### Here's what I wanted or expected to see:
Nothing.  Both were explicitly disabled.

#### Why would anyone want to do this?

Because I use shellcheck to check my zsh code.  It's better to have a few scattered `disable` directives than broken zsh code!

**Comment by dimo414** [comment]
You didn't include a shebang, so ShellCheck can't know that you're asking it to parse zsh. If you do so, you'll see a different message:

<img width="574" alt="Screen Shot 2020-04-27 at 9 26 18 AM" src="https://user-images.githubusercontent.com/1885701/80396289-493e8880-8869-11ea-8cf4-40c22daad341.png">

See [SC1071](https://github.com/koalaman/shellcheck/wiki/SC1071) and #809.

**Comment by HaleTom** [comment]
@dimo414 thanks for pointing this out.

The file causing the issue actually has:

```
#!/bin/bash
#      ^----- get shellcheck hints based on bash
# https://github.com/koalaman/shellcheck/issues/809
```

I've updated the top post

**Comment by dimo414** [comment]
I'm not sure what exactly you wanted to happen if you're using zsh syntax in a file with a `#!/bin/bash` shebang, but ShellCheck is correctly warning you that the contents you've pasted is not valid bash.

**Comment by HaleTom** [comment]
@dimo414 Actually, ShellCheck is **in**correctly warning me that the contents I've pasted is not valid bash.

I **explicitly** told it to disable emitting those warnings, and it did so anyway.

Shellcheck is very useful for catching other issues in the file that I'm `source`ing, which is why I'm using it.

**Comment by mnuccioarpae** [comment]
@HaleTom The warning says:

    SC1073: Couldn't parse this explicit subshell. **Fix to allow more checks.**

Maybe a parsing error cannot be muted?

**Comment by dimo414** [comment]
I really don't think you should expect _anything_ to work correctly if you're going to hand ShellCheck a bunch of zsh and claim it's bash. That this strategy worked at all for you is coincidental.

**Comment by koalaman** [comment]
This happened because the parse errors were appended after parsing had terminated with failure, and therefore weren't subject to the same annotation filtering as errors emitted *during* parsing. This has now been fixed.

Be aware that when SC1073 says "Fix to allow more checks", it refers to the fact that it failed to parse a syntax tree on which it can run the semantic level checks (SC2xxx). This does not change when the error is cosmetically hidden.

**Comment by HaleTom** [comment]
Thanks a bunch for this improvement, @koalaman.
